kong/router.lua

```diff
@@ -504,7 +504,23 @@ local function marshall_route(r)
       for i = 1, count do
         local path = paths[i]
 
-        if re_find(path, [[[a-zA-Z0-9\.\-_~/%]*$]], "ajo") then
+        -- we are supporting boths 2.x route path and 3.0 route paths
+        -- for a path not starting with ~, we assume it is a 2.x route path
+        -- it's safe because even if it's a 3.0 prefix path, we can normalize a prefix path
+        -- more than once and get the same result
+        -- and for a path start with ~ we just do what 3.0 do
+        local is_not_regex
```

Comment:
I'd change `is_not_regex` to `is_regex`, initialize it to `true, lose the `not not` and change the condition to `if not is_regex then`.  Embedded logic inversion in variable names makes code harder to understand.

Reply:
The `not not` is to convert the result into a boolean.
2 solutions:
1. to rename the variable to `is_prefix`;
2. or change the logic to `is_not_regex`.